src/test_board.py

```diff
@@ -114,8 +116,43 @@ def test_add_marble_to_board_raise_exception(self, position, board):
 
 
         self.assertTrue("Position given is not correct" in str(context.exception))
+    
+    good_rotation_values = lambda: (
+        ( "1", generate_board_and_add_position((0, 0)), generate_board_and_add_position((2, 0)) ),
+        ( "2", generate_board_and_add_position((0, 0)), generate_board_and_add_position((0, 2)) ),
+        ( "3", generate_board_and_add_position((0, 3)), generate_board_and_add_position((2, 3)) ),
+        ( "4", generate_board_and_add_position((0, 3)), generate_board_and_add_position((0, 5)) ),
+        ( "5", generate_board_and_add_position((3, 3)), generate_board_and_add_position((5, 3)) ),
+        ( "6", generate_board_and_add_position((3, 3)), generate_board_and_add_position((3, 5)) ),
+        ( "7", generate_board_and_add_position((3, 0)), generate_board_and_add_position((5, 0)) ),
+        ( "8", generate_board_and_add_position((3, 0)), generate_board_and_add_position((3, 2)) ),
+    )
+
+    @data_provider(good_rotation_values)
+    def test_rotate_quarter_of_board_should_return_board(self, user_value, board, expected_board):
+        print_board_if_verbosity_is_set(board)
+        print_board_if_verbosity_is_set(expected_board)
+        result = rotate_quarter_of_board(board, user_value)
+
+        np.testing.assert_array_equal(result, expected_board)
+
+
+    bad_rotation_values = lambda: (
+        ( "0", generate_empty_board()),
+        ( "A1", generate_empty_board()),
+        ( "something", generate_empty_board()),
+        ( "9", generate_empty_board()),
+    )
+    @data_provider(bad_rotation_values)
+    def test_rotate_quarter_of_board_raise_exception(self, user_value, board):
+
+        print_board_if_verbosity_is_set(board)
+        
+        with self.assertRaises(ValueError) as context:
+            board = rotate_quarter_of_board(board, user_value)
 
 
+        self.assertTrue("Rotation given is not correct" in str(context.exception))
```

Comment:
if you compare to string do not use true but an equals assertion, because if the test fails you will get a expected true but was false.